opensearchpy/helpers/signer.py

```diff
@@ -0,0 +1,128 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+#
+# Modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+#
+#  Licensed to Elasticsearch B.V. under one or more contributor
+#  license agreements. See the NOTICE file distributed with
+#  this work for additional information regarding copyright
+#  ownership. Elasticsearch B.V. licenses this file to you under
+#  the Apache License, Version 2.0 (the "License"); you may
+#  not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+# 	http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+
+import sys
+
+import requests
+
+from ..exceptions import ImproperlyConfigured
+
+OPENSEARCH_SERVICE = "es"
+
+
+def python_version_gte_36():  # type: ignore
+    """
+    A util method to check python version.
+    :return: boolean that returns true if python version running is greater than
+    equal to 3.6 otherwise returns false
+    """
+    if sys.hexversion >= 0x03060000:
+        return True
+    else:
+        return False
+
+
+try:
+    if python_version_gte_36():  # type: ignore
+        from urllib.parse import parse_qs, urlencode, urlparse
+except ImportError:
+    raise ImproperlyConfigured(
+        "Please upgrade the python version to 3.6 or above to use AWS Signer."
+    )
```

Comment:
Should the entire client require 3.6 or newer?

Reply:
At the moment, 1.x version contains support for both python 2.x and 3.x. Unfortunately, SigV4 from botocore recommends only python 3.x support. If we want to drop python 2.x support, we can do it only in next major version change like opensearch-py 2.0.
I see two options,
1. Create 2.x branch, drop python 2.x support and include this feature only in 2.x
2. Have this feature in 1.x with catch import error like Django.

Reply:
Option 1 sounds good to me @VijayanB